extensions/amp-embedly/0.1/amp-embedly.js

```diff
@@ -0,0 +1,190 @@
+ */
+export class AmpEmbedly extends AMP.BaseElement {
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @const @private {!../../../src/service/xhr-impl.Xhr} */
+    this.xhr_ = Services.xhrFor(this.win);
+
+    /** @private {?string}  */
+    this.key_ = null;
+
+    /** @private {?string}  */
+    this.url_ = null;
+
+    /** @private {?Element} */
+    this.iframe_ = null;
+  }
+
+  /** @override */
+  buildCallback() {
+    this.key_ = user().assert(
+        this.element.getAttribute('data-key'),
+        `The data-key attribute is required for <${TAG}> %s`,
+        this.element
+    );
+
+    this.url_ = user().assert(
+        this.element.getAttribute('data-url'),
+        `The data-url attribute is required for <${TAG}> %s`,
+        this.element
+    );
+
+    return this.getOembedData_()
+        .then(this.getIframe_.bind(this))
+        .then(iframe => {
+          this.iframe_ = iframe;
+          this.element.appendChild(iframe);
+
+          return this.loadPromise(iframe);
+        });
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /**
+   * Fetches oEmbed data from embedly's api.
+   * @return {!Promise}
+   * */
+  getOembedData_() {
+    const params = dict({'key': this.key_, 'url': this.url_});
+    const url = addParamsToUrl(BASE_API_URL, params);
+
+    return this.xhr_.fetchJson(url, {
+      requireAmpResponseSourceOrigin: false,
+    }).then(res => res.json());
+  }
+
+  /**
+   * Gets component iframe with set source based on data type.
+   *
+   * For RICH and VIDEO types, when oEmbed response contains:
+   * - html iframe tag: we render its source as the src of our 3p iframe.
+   * - html with a script tag: we render the html and add the script to the
+   *   3p iframe document.
+   *
+   * @param {!JsonObject<{type: string, html: string, url: string}>} data
+   * @returns {Element}
+   * @private
+   */
+  getIframe_(data) {
+    const iframe = getIframe(this.win, this.element, 'embedly');
+    this.applyFillContent(iframe);
+
+    /** @type {string} */
+    let src;
+
+    switch (data['type']) {
+      case resourceType.VIDEO:
+      case resourceType.RICH: {
+        const match = data['html'].match(SRC_REGEXP);
+
+        user().assert(
+            match, `src not found in embedly response: "${data['html']}"`
+        );
+
+        const srcUrl = `https:${match[1]}`;
+
+        if (data['html'].indexOf('<iframe') !== -1) {
+          src = srcUrl;
+          break;
+        }
+
+        src = this.createObjectUrl_(data['html']);
+
+        iframe.onload = function() {
+          const win = this.contentWindow.window;
+          const script = win.document.createElement('script');
+          script.src = srcUrl;
+          win.document.body.appendChild(script);
+
+          iframe.readyState = 'complete';
+        };
+
+        break;
+      }
+
+      case resourceType.PHOTO: {
+        const html = `
```

Comment:
Please write this on a single line, or concatenate strings without any whitespace. Otherwise the empty whitespace will be distributed in the binary, we don't want that ;)